app/mm-redux/actions/integrations.ts

```diff
@@ -399,3 +410,51 @@ export function submitInteractiveDialog(submission: DialogSubmission): ActionFun
         return {data};
     };
 }
+
+export function handleGotoLocation(href: string, intl: any): ActionFunc {
+    return async (dispatch: DispatchFunc, getState: GetStateFunc) => {
+        const state = getState();
+        const config = getConfig(state);
+
+        const {url, match} = await getURLAndMatch(href, getCurrentUrl(state), config.SiteURL);
+
+        if (match) {
+            switch (match.type) {
+            case DeepLinkTypes.CHANNEL:
+                dispatch(handleSelectChannelByName(match.channelName, match.teamName, () => DraftUtils.errorBadChannel(intl)));
+                break;
+            case DeepLinkTypes.PERMALINK:
+                dispatch(loadChannelsByTeamName(match.teamName, () => permalinkBadTeam(intl)));
```

Comment:
if this fails, we are showing the alert and then the permalink view, is that what we want ?